lib/service/Service.ts

```diff
@@ -170,7 +170,32 @@ class Service extends EventEmitter {
     argChecks.HAS_HOST({ host });
     argChecks.VALID_PORT({ port });
 
+    const peerActive = new Promise((resolve) => {
+      this.pool.once('peer.active', (peerPubKey) => {
+        if (peerPubKey === nodePubKey) {
+          resolve();
+        }
+      });
+    });
+
+    const peerClose: Promise<string> = new Promise((resolve) => {
+      this.pool.once('peer.close', (peerPubKey, rejectionMsg) => {
+        if (peerPubKey === nodePubKey) {
+          resolve(rejectionMsg);
+        }
+      });
+    });
+
+    const timeout: Promise<void> = new Promise((resolve) => {
+      setTimeout(resolve, 1000);
+    });
+
     await this.pool.addOutbound({ host, port }, nodePubKey, retryConnecting, true);
+    await peerActive;
```

Comment:
If we never get the `peer.active` event, will we proceed past this line? I suspect not although I may be reading this wrong.

I was thinking you could combine the `peerActive` and `peerClose` promises into a single promise that resolves if it gets the active event and rejects if it gets the closed event. You could also put the timeout in there and have it reject if the timeout is reached.

Reply:
@sangaman `peer.active` will always be triggered first. Although the peer which is banned will receive `peer.close` later. So put active and close in one Promise will always get resolved.

Reply:
No I don't think `peer.active` is triggered if the peer is banned or if we don't complete the connection for any reason, we have logic right before where we emit `peer.active` to make these checks. So I still think we won't be able to proceed past this line when a peer is banned.

Reply:
@sangaman We got `peer.active` on the banned peer because the ban-assertion on it only checks it's own ban list. But our case is it is banned by another peer. So the fact is the other peer's ban-assertion works and `peer.active` will not be emitted. But on our side it will bypass the peer ban-assertion and emit `peer.active`